pkg/controllers/jobset_controller.go

```diff
@@ -673,10 +669,21 @@ func labelAndAnnotateObject(obj metav1.Object, js *jobset.JobSet, rjob *jobset.R
 	annotations[jobset.JobIndexKey] = strconv.Itoa(jobIdx)
 	annotations[jobset.JobKey] = jobHashKey(js.Namespace, jobName)
 
-	// Only set exclusive key label/annotation on jobs/pods if the parent JobSet
-	// is using exclusive placement.
-	if _, exists := js.Annotations[jobset.ExclusiveKey]; exists {
-		annotations[jobset.ExclusiveKey] = js.Annotations[jobset.ExclusiveKey]
+	// Check for JobSet level exclusive placement.
+	if topologyDomain, exists := js.Annotations[jobset.ExclusiveKey]; exists {
+		annotations[jobset.ExclusiveKey] = topologyDomain
+		// Check if we are using nodeSelectorStrategy implementation of exclusive placement at the JobSet level.
+		if value, ok := js.Annotations[jobset.NodeSelectorStrategyKey]; ok {
+			annotations[jobset.NodeSelectorStrategyKey] = value
+		}
+	}
+	// Check for ReplicatedJob level exclusive placement.
+	if topologyDomain, exists := rjob.Template.Annotations[jobset.ExclusiveKey]; exists {
+		annotations[jobset.ExclusiveKey] = topologyDomain
+		// Check if we are using nodeSelectorStrategy implementation of exclusive placement at the ReplicatedJob level.
+		if value, ok := rjob.Template.Annotations[jobset.NodeSelectorStrategyKey]; ok {
```

Comment:
This labelAndAnnotate function is invoked for both the job template and the pod template. The job template inherits gets this already because the job is created out of `rjob.Template` anyways, and we don't need to set this annotation on the pod.

Reply:
anyways, we can clean this up when we add the canonical placement policy API.